core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -438,7 +440,7 @@ class GroupCoordinator(val brokerId: Int,
           case None => group.currentState match {
             case PreparingRebalance =>
               val member = group.get(memberId)
-              updateMemberAndRebalance(group, member, protocols, s"Member ${member.memberId} joining group during ${group.currentState}; client reason: $reason", responseCallback)
+              updateMemberAndRebalance(group, member, protocols, rebalanceTimeoutMs, sessionTimeoutMs, s"Member ${member.memberId} joining group during ${group.currentState}; client reason: $reason", responseCallback)
```

Comment:
Updating the rebalance timeout and the session timeout in the non-static cases is not really necessary. I suppose that it does not hurt to do it.